server/src/main/java/org/elasticsearch/index/mapper/DataStreamTimestampFieldMapper.java

```diff
@@ -34,10 +34,17 @@
     public static final String NAME = "_data_stream_timestamp";
     private static final String DEFAULT_PATH = "@timestamp";
 
+    private static final DataStreamTimestampFieldMapper ENABLED_INSTANCE =
+        new DataStreamTimestampFieldMapper(TimestampFieldType.INSTANCE, true);
+    private static final DataStreamTimestampFieldMapper DISABLED_INSTANCE =
+        new DataStreamTimestampFieldMapper(TimestampFieldType.INSTANCE, false);
+
     // For now the field shouldn't be useable in searches.
     // In the future it should act as an alias to the actual data stream timestamp field.
     public static final class TimestampFieldType extends MappedFieldType {
 
+        static final TimestampFieldType INSTANCE = new TimestampFieldType();
+
         public TimestampFieldType() {
```

Comment:
Shall we make this constructor private since we have a singleton instance available now?

Reply:
++ makes sense :)